hive/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -101,8 +101,8 @@ private TServer newThriftServer(TServerSocket socket, HiveConf conf) throws Exce
         .processor(new TSetIpAddressProcessor<>(handler))
         .transportFactory(new TTransportFactory())
         .protocolFactory(new TBinaryProtocol.Factory())
-        .minWorkerThreads(3)
-        .maxWorkerThreads(5);
+        .minWorkerThreads(10)
+        .maxWorkerThreads(15);
```

Comment:
Why did this change? I don't think these should be modified because these ensure that Iceberg doesn't leak connections to the metastore.

Reply:
Thats a good point. Maybe there is a leak in SparkTableUtil.partitionDFByFilter() which was causing multiple calls to this method in testPartitionScanByFilter() to fail, without increasing the number of worker threads.

I will investigate that more.
thanks,

Reply:
I think the problem is the connection used in `hacks.Hive`. Now that we have a metastore client in `iceberg-hive`, we should use that instead.

Reply:
Using a metastore client (HiveMetastoreClient), i believe will lead to a major rewrite of the SparkTableUtil.java class. Could that be a different change ?